Core/Src/system.c

```diff
@@ -79,22 +96,53 @@ void system_main(System *sys) {
 
 void commandHandler(System *sys) {
 	uint8_t command;
-	uint8_t status_error = pop_ring_buffer(sys->recieve_ring_buffer, &command);
+	uint8_t status_error = pop_ring_buffer(sys->receive_ring_buffer, &command);
 	if (!status_error) {
-		switch (command) {
-		case LED1_5Hz:
-			sys->LED1_timer = sys->h_timer1;
-			break;
-		case LED1_1Hz:
-			sys->LED1_timer = sys->h_timer2;
-			break;
-		case LED2_5Hz:
-			sys->LED2_timer = sys->h_timer1;
-			break;
-		case LED2_1Hz:
-			sys->LED2_timer = sys->h_timer2;
-			break;
+
+		if (command == sys->LED1_State) {
+			sys->LED1_State = OFF;
+			HAL_GPIO_WritePin(sys->h_GPIO_LED, sys->LEDPin1, GPIO_PIN_RESET);
+			HAL_UART_Transmit_IT(&huart3, (uint8_t*)"Turning LED1 OFF.\n", sizeof("Turning LED1 OFF.\n"));
+
+		}
+		else if (command == sys->LED2_State) {
+			sys->LED2_State = OFF;
+			HAL_GPIO_WritePin(sys->h_GPIO_LED, sys->LEDPin2, GPIO_PIN_RESET);
+			HAL_UART_Transmit_IT(&huart3, (uint8_t*)"Turning LED2 OFF.\n", sizeof("Turning LED2 OFF.\n"));
+
+		}
+		else {
+			switch (command) {
```

Comment:
I appreciate the use of switch cases.  I tend to avoid them in other languages, but they are invaluable in C, so good inclusion here